TestsOutsidePackage/test_score.py

```diff
@@ -54,3 +93,272 @@ def test_run_scoring(test_output_dirs: OutputFolderForTests, is_ensemble: bool)
+    model_config.set_output_to(test_output_dirs.root_dir)
+    checkpoint_path = model_config.checkpoint_folder / "checkpoint.ckpt"
+    create_model_and_store_checkpoint(model_config, checkpoint_path)
+
+    azure_config = AzureConfig()
+    project_root = Path(__file__).parent.parent
+    ml_runner = MLRunner(model_config=model_config, azure_config=azure_config, project_root=project_root)
+    model_folder = test_output_dirs.root_dir / "final"
+    ml_runner.copy_child_paths_to_folder(model_folder=model_folder, checkpoint_paths=[checkpoint_path])
+
+    score_pipeline_config = ScorePipelineConfig(
+        data_folder=TEST_DATA_DIR,
+        model_folder=str(model_folder),
+        image_files=[str(HN_DICOM_SERIES_ZIP)],
+        result_image_name=HNSEGMENTATION_FILE.name,
+        use_gpu=False,
+        use_dicom=True)
+
+    with mock.patch('score.run_inference',
+                    return_value=mock_segmentation) as mock_run_inference:
+        with mock.patch('score.store_as_ubyte_nifti',
+                        return_value=HNSEGMENTATION_FILE) as mock_store_as_ubyte_nifti:
+            segmentation = score_image(score_pipeline_config)
+            assert_zip_file_contents(segmentation, HN_DICOM_RT_ZIPPED, model_folder)
+
+    mock_run_inference.assert_called()
+    mock_store_as_ubyte_nifti.assert_called()
+
+
+def test_score_image_dicom_mock_run(test_output_dirs: OutputFolderForTests) -> None:
+    """
+    Test that dicom in and dicom-rt out works, by mocking out only the run scoring function.
+
+    :param test_output_dirs: Test output directories.
+    """
+    model_config = DummyModel()
+    model_config.set_output_to(test_output_dirs.root_dir)
+    checkpoint_path = model_config.checkpoint_folder / "checkpoint.ckpt"
+    create_model_and_store_checkpoint(model_config, checkpoint_path)
+
+    azure_config = AzureConfig()
+    project_root = Path(__file__).parent.parent
+    ml_runner = MLRunner(model_config=model_config, azure_config=azure_config, project_root=project_root)
+    model_folder = test_output_dirs.root_dir / "final"
+    ml_runner.copy_child_paths_to_folder(model_folder=model_folder, checkpoint_paths=[checkpoint_path])
+
+    score_pipeline_config = ScorePipelineConfig(
+        data_folder=TEST_DATA_DIR,
+        model_folder=str(model_folder),
+        image_files=[str(HN_DICOM_SERIES_ZIP)],
+        result_image_name=HNSEGMENTATION_FILE.name,
+        use_gpu=False,
+        use_dicom=True)
+
+    image_with_header = io_util.load_nifti_image(HNSEGMENTATION_FILE)
+
+    with mock.patch('score.run_inference',
+                    return_value=image_with_header.image) as mock_run_inference:
+        segmentation = score_image(score_pipeline_config)
+        assert_zip_file_contents(segmentation, HN_DICOM_RT_ZIPPED, model_folder)
+
+    mock_run_inference.assert_called()
+
+
+def test_score_image_dicom_mock_none(test_output_dirs: OutputFolderForTests) -> None:
+    """
+    Test that dicom in and dicom-rt out works.
```

Comment:
This is the test that runs inferencing end-to-end, right? How long does that run?

Reply:
Takes about 45 seconds on my laptop.